app/models/congress_member.rb

```diff
@@ -201,12 +210,29 @@ def fill_out_form_with_capybara f={}, driver
         when "fill_in"
           if a.value.starts_with?("$")
             if a.value == "$CAPTCHA_SOLUTION"
-              location = CAPTCHA_LOCATIONS.keys.include?(bioguide_id) ? CAPTCHA_LOCATIONS[bioguide_id] : session.driver.evaluate_script('document.querySelector("' + a.captcha_selector.gsub('"', '\"') + '").getBoundingClientRect();')
-
-              url = self.class::save_captcha_and_store_poltergeist session, location["left"], location["top"], location["width"], location["height"]
+              if a.options and a.options["google_recaptcha"]
+                begin
+                  url = self.class::save_google_recaptcha_and_store_poltergeist(session,a.captcha_selector)
+                  captcha_value = yield url
+                  session.within_frame(recaptcha_frame_index(session)) do
+                    for i in captcha_value.split(",")
+                      session.execute_script("document.querySelector('.fbc-imageselect-checkbox-#{i}').checked=true")
+                    end
+                    sleep 0.5
+                    session.find(".fbc-button-verify input").trigger('click')
+                    @recaptcha_value = session.find("textarea").value
+                  end
+                  session.fill_in(a.name,with:@recaptcha_value)
+                rescue Exception => e
```

Comment:
This is actually a bad way to do it and I've been thinking of a better way. This is how we "loop" the recaptcha attemps. If the poltergeist session raises an exception because it failed to find  expected data (the recaptcha success response token), we retry and go back to retrieving another captcha. The rescue/retry here is clearly a problem, though. If the exception happens while retrieving the captcha, the process will be stuck forever.

I think the way to go here would be doing this inside a better handled while block, that starts with a nil recaptcha_success_token and loops until it's not nil anymore.